### PR TITLE
Shut the error logger in the tests up

### PR DESCRIPTION
**Pull request by cmcaine**
Also validate that it does log the message we expect.

### CODE REVIEW

**Comment by jpsamaroo** [comment]
Looks good! Maybe you just need to `close(mock_stderr)` to fix the `unlink` error?

File: test/runtests.jl [hunk 1]
```diff
@@ -67,6 +71,9 @@ serve(test, 8001)
 @test String(HTTP.get("http://localhost:8001/badurl";
                       status_exception=false).body) ==
              "Internal server error"
+seekstart(mock_stderr)
+@test occursin("An error!", read(mock_stderr, String))
```
Comment (cmcaine) [inline test/runtests.jl:75]:
```suggestion
@test occursin("An error!", read(mock_stderr, String))
close(mock_stderr)
```

**Comment by cmcaine** [comment]
This is good to merge now :)

**Comment by jpsamaroo** [comment]
Thanks!

**Pull request merged**